src/main/java/com/hivemq/cli/mqtt/AbstractMqttClientExecutor.java

```diff
@@ -508,4 +508,10 @@ public boolean isConnected(final @NotNull Context context) {
                     new String(mqtt3Publish.getPayloadAsBytes(), StandardCharsets.UTF_8));
         }
     }
+
+    private @NotNull String getClientDataKey(final @NotNull MqttClient client) {
+        final String clientId = client.getConfig().getClientIdentifier().map(Objects::toString).orElse("null");
+        final String serverHost = client.getConfig().getServerHost();
+        return "client {" + "identifier='" + clientId + '\'' + ", host='" + serverHost + '\'' + '}';
```

Comment:
```suggestion
        return "client {identifier='" + clientId + "', host='" + serverHost + "'}";
```

Reply:
In line 343 we're using this code to generate the key:
```
        final String key = MqttUtils.buildKey(client.getConfig().getClientIdentifier().map(Object::toString).orElse(""),
                client.getConfig().getServerHost());
```
I guess that's even better to not duplicate the logic.